pkg/spec/spec.go

```diff
@@ -167,18 +178,36 @@ func ExpandEnv(s string) (string, error) {
 	}
 
 	s = escapeString(s, preventEscapingPrefix)
-	s = escapeFleetVar(s, preventEscapingPrefix)
 	var err *multierror.Error
-	s = os.Expand(s, func(env string) string {
-		if strings.HasPrefix(env, preventEscapingPrefix) {
-			return "$" + strings.TrimPrefix(env, preventEscapingPrefix)
+	s = fleet.MaybeExpand(s, func(env string) (string, bool) {
+		switch {
+		case strings.HasPrefix(env, preventEscapingPrefix):
+			return "$" + strings.TrimPrefix(env, preventEscapingPrefix), true
```

Comment:
We should return the variable in brackets in case it was already in brackets. It doesn't hurt
```suggestion
			return "${" + strings.TrimPrefix(env, preventEscapingPrefix) + "}", true
```

Reply:
The thing is that this is not a variable. This logic is for escaping a variable. Lucas added a lot of tests for these, like:

<img width="958" alt="image" src="https://github.com/user-attachments/assets/49541076-c8fc-4c87-90bb-d282ec80456f" />